pkg/providers/instance/azure_client.go

```diff
@@ -110,32 +128,52 @@ func NewAZClient(cfg *auth.Config, env *azure.Environment) (*AZClient, error) {
 	}, nil
 }
 
-func setArmClientOptions() *arm.ClientOptions {
-	opt := new(arm.ClientOptions)
-
-	opt.PerCallPolicies = append(opt.PerCallPolicies,
-		PolicySetHeaders{
-			"Referer": []string{RPReferer},
-		},
-		PolicySetHeaders{
-			"x-ms-correlation-request-id": []string{uuid.New().String()},
+func setE2eArmClientOptions(cfg *auth.Config, transporter *http.Client) *arm.ClientOptions {
+
+	cloudConfig := getCloudConfiguration(cfg.CloudEnvironment, true)
+
+	return &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			PerCallPolicies: []policy.Policy{
+				PolicySetHeaders(http.Header{
+					"Referer": []string{
+						"https://" + auth.E2E_RP_INGRESS_ENDPOINT_ADDRESS,
+					},
+					"x-ms-correlation-request-id": []string{uuid.New().String()},
+					"x-ms-home-tenant-id":         []string{cfg.TenantID},
+				}),
+			},
+			Transport: transporter,
+			Logging: policy.LogOptions{
+				IncludeBody: true,
+			},
+			Cloud: cloudConfig,
 		},
-	)
-	opt.Cloud.Services = maps.Clone(opt.Cloud.Services) // we need this because map is a reference type
-	opt.Cloud.Services[cloud.ResourceManager] = cloud.ServiceConfiguration{
-		Audience: cloud.AzurePublic.Services[cloud.ResourceManager].Audience,
-		Endpoint: "https://" + RPReferer,
 	}
-	return opt
 }
 
-// PolicySetHeaders sets http header
-type PolicySetHeaders http.Header
+func getCloudConfiguration(cloudName string, e2eMode bool) cloud.Configuration {
+	var cloudConfig cloud.Configuration
+	switch strings.ToLower(cloudName) {
+	case string(AzurePublicCloud):
+		cloudConfig = cloud.AzurePublic
+	case string(AzureUSGovernmentCloud):
+		cloudConfig = cloud.AzureGovernment
+	case string(AzureChinaCloud):
+		cloudConfig = cloud.AzureChina
+	default:
+		panic("cloud config does not exist")
```

Comment:
how about use `AzurePublicCloud` as the default routine?  so we can keep the consistency with the previous versions which don't support CLOUD_ENVIRONMENT env.

Reply:
addressed.